app/src/main/java/de/westnordost/streetcomplete/osm/Maxspeed.kt

```diff
@@ -10,6 +10,17 @@ val MAXSPEED_TYPE_KEYS = setOf(
     "zone:traffic"
 )
 
+val anyMaxSpeedTagKey = "~${(MAXSPEED_TYPE_KEYS + "maxspeed").joinToString("|")}"
+
+val explicitlyInSlowZone = """$anyMaxSpeedTagKey ~ ".*:(zone:?)?([1-9]|[1-2][0-9]|30)""""
+
+val notInSlowZoneAndNotPlainMaxspeedTagging = """
+  !(
+    maxspeed ~ "walk|(\d+( ?mph)?)" and !~${(MAXSPEED_TYPE_KEYS).joinToString("|")}
```

Comment:
uhm, but what about `source:maxspeed=sign` then?

Reply:
(Also, is there a test that additionally checks for this condition? I am having still problems undestanding why this is there in the first place, and a test would also serve as documentation.)

Reply:
> If I leave out the part about having only the `maxspeed` tag and no other maxspeed tags, the following test fails. That means the quest would be shown for more elements than desired:

> https://github.com/streetcomplete/StreetComplete/blob/e300eaab151c413c387e9a9e936e2b9013ad5dab/app/src/test/java/de/westnordost/streetcomplete/quests/lanes/AddLanesTest.kt#L55-L59

How can it fail, though? The filter currently looks like this:

```
or highway = residential and (
             maxspeed > 33
             or ~"${(MAXSPEED_TYPE_KEYS + "maxspeed").joinToString("|")}" ~ ".*:(urban|rural|trunk|motorway|nsl_single|nsl_dual)"
           )
```

... see the maxspeed > 33

Reply:
The current test doesn't fail.
But it fails, if I leave out `maxspeed ~ "walk|(\d+( ?mph)?)" and !~${(MAXSPEED_TYPE_KEYS).joinToString("|")}`  the Because it's not explicitly in a slow zone `!($explicitlyInSlowZone)` becomes true.